master/internal/rm/kubernetesrm/job.go

```diff
@@ -143,284 +140,363 @@ func newPod(
 		masterTLSConfig:      masterTLSConfig,
 		loggingTLSConfig:     loggingTLSConfig,
 		loggingConfig:        loggingConfig,
-		slots:                msg.Slots,
+		numPods:              msg.NumPods,
+		slotsPerPod:          msg.Slots,
```

Comment:
conceptual question, why are we tracking down the pod level, even with jobs deployments?

Reply:
we still need to form a pod spec and the pod spec makes the resource requests. e.g., user submits experiment with slots_per_trial =16 and we know we do max_slots_per_pod = 4, then we need to launch 1 job with parallelism = 4, completions = 4, and 4 slots per pod (so `nvidia.com/gpu: 4`).